zap/src/main/java/org/parosproxy/paros/network/HttpMessage.java

```diff
@@ -990,6 +990,11 @@ public void mutateHttpMethod(String method) {
             if (prevMethod.equalsIgnoreCase(method)) {
                 return;
             }
+            // When changing methods, remove the content-type if a method should not have a body
+            // This does not prevent the user for modifying the request afterwards if needed
+            if (!(method.equals(HttpRequestHeader.POST) || method.equals(HttpRequestHeader.PUT))) {
```

Comment:
IMO we should only change the Content-Type if we are actually changing the parameters around. (This also misses methods where no body is expected.)

Reply:
I am not sure if I understood the comment here. 
I have put this check because if we are switching from a POST or PUT back to a GET (or any other method that does not have a body), there should be no Content-Type set, since all the parameters are moved back as query parameters. That's why by setting it to null we are removing it.

Reply:
Mutating from PUT to GET does not change anything. The body is not changed unless mutating from/to POST, so we shouldn't be changing unless it's those cases.

Reply:
Thanks for the clarification, it makes sense. I will just remove the content-type if the previous method is POST then